src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/MultiUserAKVTest.java

```diff
@@ -0,0 +1,643 @@
+            // which contains the required provider
+            try (SQLServerPreparedStatement pstmt2 = (SQLServerPreparedStatement) TestUtils.getPreparedStmt(con, sql,
+                    SQLServerStatementColumnEncryptionSetting.Enabled)) {
+                pstmt2.setInt(1, customerId);
+                
+                try {
+                    ResultSet rs = pstmt2.executeQuery(); 
+                    fail(TestResource.getResource("R_expectedExceptionNotThrown"));
+                } catch (Exception ex) {
+                    // kz debug
+                    System.out.println(ex.getMessage());
+                    assertTrue(ex instanceof UnsupportedOperationException);
+                }
+            }
+        } finally {
+            dropObject(AETestConnectionString, "TABLE", customProviderTableName);
+            dropObject(AETestConnectionString, "CEK", cekDummy);
+            dropObject(AETestConnectionString, "CMK", cmkDummy);
+
+            SQLServerConnection.unregisterColumnEncryptionKeyStoreProviders();
+        }
+    }
+
+    private void insertData(String tableName, int customId, String customName) {
+        String sqlQuery = "INSERT INTO " + tableName + " VALUES ( ?, ? )"; 
+
+        try (SQLServerConnection con = PrepUtil.getConnection(AETestConnectionString, AEInfo);
+                SQLServerPreparedStatement pstmt = (SQLServerPreparedStatement) TestUtils.getPreparedStmt(con, sqlQuery,
+                    SQLServerStatementColumnEncryptionSetting.Enabled)) { 
+            pstmt.setInt(1, customId);
+            pstmt.setString(2, customName);
+            pstmt.executeUpdate();
+            System.out.println("1 Record inserted successfully.");                              
+        } catch(SQLException e) {
+            fail(e.getMessage());
+        }        
+    }
+
+    private void createTableForCustomProvider(String connString, String tableName, String cekName) {
+        String sqlQuery = "CREATE TABLE " + tableName + " (" 
+                + " [CustomerId] [int] ENCRYPTED WITH (COLUMN_ENCRYPTION_KEY = [" + cekName + "], ENCRYPTION_TYPE = DETERMINISTIC, ALGORITHM = 'AEAD_AES_256_CBC_HMAC_SHA_256'), "
+                + " [CustomerName] [varchar](50) COLLATE Latin1_General_BIN2 ENCRYPTED WITH (COLUMN_ENCRYPTION_KEY = [" + cekName + "], ENCRYPTION_TYPE = DETERMINISTIC, ALGORITHM = 'AEAD_AES_256_CBC_HMAC_SHA_256') "
+                + ")";
+
+        try (SQLServerConnection con = PrepUtil.getConnection(connString, AEInfo);
+                SQLServerStatement stmt = (SQLServerStatement) con.createStatement()) { 
+            TestUtils.dropTableIfExists(tableName, stmt);
+            stmt.execute(sqlQuery);                              
+        } catch(SQLException e) {
+            fail(e.getMessage());
+        }        
+    }
+
+    private void dropObject(String connString, String objectType, String objectName) {

```

Comment:
add this to TestUtils? (there is a dropObjectIfExists)

Reply:
There are some functions like dropCEK / dropCMK from package `com.microsoft.sqlserver.jdbc.AlwaysEncrypted`. I have moved this to AESetup.java.